neo/SmartContract/ApplicationEngine.cs

```diff
@@ -9,8 +9,8 @@ namespace Neo.SmartContract
 {
     public partial class ApplicationEngine : ExecutionEngine
     {
-        public static event EventHandler<NotifyEventArgs> Notify;
-        public static event EventHandler<LogEventArgs> Log;
+        public event EventHandler<NotifyEventArgs> Notify;
+        public event EventHandler<LogEventArgs> Log;
 
         public const long GasFree = 0;
```

Comment:
I think that this change is good for unit testing, because we should try to avoid static objects and events. What do you think @neo-project/core

Reply:
Ping @neo-project/core @neo-project/developers

Reply:
Looks ok if its needed.

In the future, we can ensure it const (readonly) under its use on some contexts.

Reply:
Fully agree @shargon ! Otherwise we cannot build two ApplicationEngine.... how did we do that??? This change is fundamental.

Reply:
No. If you do this, you can not listen to the Verification events.

Reply:
I think that Is better to add the mandatory calls to this event in the constructor. Static is bad for UT hahahah